src/AfricasTalking.php

```diff
@@ -24,17 +23,11 @@ class AfricasTalking
 
 	public function __construct($username, $apiKey)
 	{
-		if($username === 'sandbox') {
-			$this->baseDomain = self::BASE_SANDBOX_DOMAIN;
-		} else {
-			$this->baseDomain = self::BASE_DOMAIN;
-		}
-
-		$this->baseUrl = "https://api." . $this->baseDomain . "/version1/";
-		$this->voiceUrl = "https://voice." . $this->baseDomain . "/";
-		$this->paymentsUrl = "https://payments." . $this->baseDomain . "/";
-		$this->contentUrl = ($username === "sandbox") ? ($this->baseUrl) : ("https://content." . $this->baseDomain . "/version1/");
-		$this->checkoutTokenUrl = "https://api." . $this->baseDomain . "/";
+		$this->baseUrl = "https://api." . $this->base_domain() . "/version1/";
```

Comment:
`$this->username` used in `base_domain()` is not set at the time the method is invoked. Consider moving https://github.com/AfricasTalkingLtd/africastalking-php/blob/a917b4cf84eaf759f7bdeff02ddcb0537e6d7fc6/src/AfricasTalking.php#L36-L37 to above this line otherwise sandbox will not work!